test/helpers/utils.go

```diff
@@ -247,6 +247,24 @@ func (kubectl *Kubectl) WaitCiliumEndpointReady(podFilter string, k8sNode string
 	return ciliumPod, endpoints
 }
 
+// WaitKafkaBroker waits for the broker to be ready, by executing
+// a produce request on existing topics and waiting for a response from broker
+func WaitKafkaBroker(pod string, cmd string, node *SSHMeta) error {
+	fmt.Print("in WaitKafkaBroker")
+	body := func() bool {
+		for {
+			res := node.ContainerExec(pod, cmd)
+			if res.success {
+				log.Info("Kafka Broker ready ")
```

Comment:
Remove the trailing space.

Reply:
This function does not make sense to me, the for loop is not needed, function should be like this: 
```
func WaitKafkaBroker(pod string, cmd string, node *SSHMeta) error {
	body := func() bool {
			res := node.ContainerExec(pod, cmd)
			return res.WasSuccessful()
		}
	}
	err := WithTimeout(body, "Kafka Broker not ready", &TimeoutConfig{Timeout: 150})
	return err
}
```

WithTimeout function, executes function with a `time.Ticker` and if return true it finished without error, it false retry after ticker. If timeout return error.

Reply:
Yes, I was planning to ping you regarding this, I had a feeling I was not using the `WithTimeout` correctly.

Reply:
Thanks.. let me make the change and get back

Reply:
done